### PR TITLE
middleware handles non-existing sources

### PR DESCRIPTION
**Pull request by Yoshemith**
This PR solves #77
I created a new folder for future middlewares which includes `existingSourceMiddleware.js`. This middleware acts as a filter between existing and non-existing news sources. What I did was, basically, I checked the list of sources and managed to create a function that finds a match with the required source, if it doesn't then proceed to redirect to a 404 not-found page.

@MizouziE, I'd appreciate it if you could review the changes or any comments about the PR.

### CODE REVIEW

File: src/middlewares/existingSourceMiddleware.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const sources = require('../data/sources.json');
+const path = require('path');
+const eSources = sources.map((source) => source.name); //Existing sources
```
Comment (MizouziE) [inline src/middlewares/existingSourceMiddleware.js:3]: The commented explanation is a good idea, but you could be even more explicit by naming the constant `existingSources`, I didn't realise the significance until I read the whole thing through a few times.

I only say this to keep it crystal clear what is happening here (which is very nice btw!)

File: src/middlewares/existingSourceMiddleware.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+function existingSourceMiddleware(req, res, next) {
+  const sourceId = req.params.sourceId;
+  const sourceFound = eSources.filter(
```
Comment (MizouziE) [inline src/middlewares/existingSourceMiddleware.js:7]: If the name is changed just be sure to change it here too

**Review by MizouziE — CHANGES_REQUESTED** [review]
Nice work! Just for the sake of keeping it all as readable as possible, can you alter the name a bit?

File: src/middlewares/existingSourceMiddleware.js (hunk 1 above)
Comment (Yoshemith) [reply src/middlewares/existingSourceMiddleware.js:3]: You're right, should've kept it simple. I'll change it right away.

**Comment by Yoshemith** [comment]
@MizouziE I changed it, let me know what you think :)

**Review by MizouziE — APPROVED** [review]
Perfect! Absolutely zero room for any confusion. Great stuff! 🔥

**Pull request merged**